tests/server/test_query_params.py

```diff
@@ -424,6 +424,47 @@ def test_list_has_and(self):
         expected_ids = ["mpf_1"]
         self._check_response(request, expected_ids, len(expected_ids))
 
+    def test_awkward_not_queries(self):
+        """ Test an awkward query from the spec examples. It should return all but 2 structures
+        in the test data. The test is done in three parts:
+
+            - first query the individual expressions that make up the OR,
+            - then do an empty query to get all IDs
+            - then negate the expressions and ensure that all IDs are returned except
+              those from the first queries.
+
+        """
+        expected_ids = ["mpf_3819"]
+        request = (
+            '/structures?filter=chemical_formula_descriptive="Ba2NaTi2MnRe2Si8HO26F"'
+        )
+        self._check_response(request, expected_ids, len(expected_ids))
+
+        expected_ids = ["mpf_2"]
+        request = '/structures?filter=chemical_formula_anonymous = "A2BC" AND NOT chemical_formula_descriptive = "Ac2AgPb" '
+        self._check_response(request, expected_ids, len(expected_ids))
+
+        request = "/structures"
+        unexpected_ids = ["mpf_3819", "mpf_2"]
+        expected_ids = [
+            struct["id"]
+            for struct in self.client.get(request).json()["data"]
+            if struct["id"] not in unexpected_ids
+        ]
+
+        print(expected_ids)
+
+        request = (
+            "/structures?filter="
+            "NOT ( "
+            'chemical_formula_descriptive = "Ba2NaTi2MnRe2Si8HO26F" AND '
+            'chemical_formula_anonymous = "A26B8C2D2E2FGHI" OR '
```

Comment:
This should be added to the query above or removed.

Reply:
Not sure what you mean, the queries above are getting the set of all structures, so that the "inverse query" test doesn't fail if we add another test case to `test_structures.json`.

Reply:
Sorry, it wasn't clear.
The query line `chemical_formula_anonymous = "A26B8C2D2E2FGHI"` should either be removed or used in the very first query in this test, I guess? At least to be rigorous and true to the spirit of the test you've written :)